code/SecureFileController.php

```diff
@@ -60,18 +61,22 @@ public function handleRequest(SS_HTTPRequest $request, DataModel $model) {
 
                 return $self->sendFile($file);
             } else {
-                if($file instanceof File) {
+                if ($file instanceof File) {
                     // Permission failure
                     Security::permissionFailure($self, 'You are not authorised to access this resource. Please log in.');
                 } else {
-                    // File doesn't exist
-                    $self->setResponse(new SS_HTTPResponse('File Not Found', 404));
+                    if (ClassInfo::class_exists('ErrorPage')) {
+                        $self->setResponse(ErrorPage::response_for(404));
+                    } else {
+                        // File doesn't exist
+                        $this->httpError(404);
```

Comment:
I'll revert the whitespace changes, but this method was mis-matched so I just made it consistent.